src/VisualStudio/CSharp/Impl/ProjectSystemShim/EntryPointFinder.cs

```diff
@@ -2,26 +2,46 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using System.Collections.Generic;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 
 namespace Microsoft.VisualStudio.LanguageServices.CSharp.ProjectSystemShim
 {
     internal class EntryPointFinder : AbstractEntryPointFinder
     {
-        protected override bool MatchesMainMethodName(string name)
-            => name == "Main";
+        private readonly INamedTypeSymbol? _task;
+        private readonly INamedTypeSymbol? _taskOf;
 
+        public EntryPointFinder(Compilation? compilation)
+        {
+            _task = compilation?.TaskType();
+            _taskOf = compilation?.TaskOfTType();
+        }
+
+        [Obsolete("FindEntryPoints on a INamespaceSymbol is deprecated, please pass in the Compilation instead.")]
         public static IEnumerable<INamedTypeSymbol> FindEntryPoints(INamespaceSymbol symbol)
```

Comment:
why not just remove these?

Reply:
see comment in the PR description: 

> NOTE: the new api added in [77e2281](https://github.com/dotnet/roslyn/pull/60614/commits/77e22811f22744d2c6b5e48631800510612bbdd5) also marks FindEntryPoints(INamespaceSymbol symbol, bool findFormsOnly). as Obsolete. Once this is merged and all upstream consumers move to the new api in https://github.com/dotnet/project-system/pull/8044 the plan is to delete the obsolete method.

Don't want to break project system so this represents an intermediate state. the Obsolete method will be removed in a follow up PR as soon as the project system PR is merged.